pkg/pillar/cmd/zedmanager/updatestatus.go

```diff
@@ -371,10 +374,92 @@ func doUpdate(ctx *zedmanagerContext, uuidStr string,
 	return changed
 }
 
-func doInstall(ctx *zedmanagerContext, uuidStr string,
-	config types.AppInstanceConfig, status *types.AppInstanceStatus) (bool, bool) {
+// doInstallProcessStorageEntriesWithVerifiedImage
+//  Returns: (vs, storageStatusUpdate)
+func doInstallProcessStorageEntriesWithVerifiedImage(
+	ctx *zedmanagerContext,
+	status *types.AppInstanceStatus,
+	ssPtr *types.StorageStatus) (*types.VerifyImageStatus, bool) {
+
+	changed := false
+	safename := ssPtr.Safename()
+
+	// Shortcut if image is already verified
+	vs := lookupVerifyImageStatusAny(ctx, safename, ssPtr.ImageSha256)
+	// Handle post-reboot verification in progress by allowing
+	// types.DOWNLOADED. If the verification later fails we will
+	// get a delete of the VerifyImageStatus and skip this
+	if vs == nil {
+		log.Debugf("Verifier status not found for %s sha %s",
+			safename, ssPtr.ImageSha256)
+		return nil, false
+	}
+	switch vs.State {
+	case types.DELIVERED:
+		log.Infof("Found verified image for %s sha %s\n",
+			safename, ssPtr.ImageSha256)
+
+	case types.DOWNLOADED:
+		log.Infof("Found downloaded/verified image for %s sha %s\n",
+			safename, ssPtr.ImageSha256)
+	default:
+		log.Infof("vs.State (%d) not DELIVERED / DOWNLOADED. safename: %s"+
+			" sha %s", vs.State, safename, ssPtr.ImageSha256)
+		return nil, false
+	}
+	if vs.RefCount == 0 {
+		log.Infof("Vs.RefCount = 0.")
+		// Check if Verifier Config exists.
```

Comment:
Perhaps adding a useful comment referring to the cleanup handshake in handleverifier.go would be more useful for the person who needs to understand this code in then future.